modules/vectra.py

```diff
@@ -669,21 +694,56 @@ def get_hosts(self, **kwargs):
             params=self._generate_host_params(kwargs),
         )
 
+    def get_threaded(self, url, count, **kwargs):
+        try:
+            page_size = kwargs["params"].pop("page_size")
+        except KeyError:
+            page_size = 5000
+        try:
+            kwargs["params"].pop("page")
+        except KeyError:
+            pass
+        pages = ceil(count / page_size)
+        with concurrent.futures.ThreadPoolExecutor(
+            max_workers=self.threads, thread_name_prefix="Get All Generator"
+        ) as executor:
+            try:
+                results = {
+                    executor.submit(
+                        self._request,
+                        method="get",
+                        url=url + f"?page={page}&page_size={page_size}",
+                        params=kwargs["params"],
+                    ): page
+                    for page in range(2, pages + 1)
+                }
+                for result in concurrent.futures.as_completed(results):
+                    yield result.result()
+            except KeyboardInterrupt:
+                executor.shutdown(wait=False, cancel_futures=True)
+
     @validate_gte_api_v3_3
     def get_all_hosts(self, **kwargs):
         """
         Generator to retrieve all hosts - all parameters are optional
         For available query params, see _generate_host_params
         """
+        url = f"{self.url}/hosts"
+        params = self._generate_host_params(kwargs)
         resp = self._request(
             method="get",
-            url=f"{self.url}/hosts",
-            params=self._generate_host_params(kwargs),
+            url=url,
+            params=params,
         )
         yield resp
-        while resp.json()["next"]:
-            resp = self._request(method="get", url=resp.json()["next"])
-            yield resp
+
+        if self.threads == 1:
```

Comment:
This may work better as a common method that is shared...

```
def yield_results(self, resp):
    if self.threads == 1:
        while resp.json()["next"]:
            resp = self._request(method="get", url=resp.json()["next"])
            yield resp
    else:
        count = resp.json()["count"]
        yield from self.get_threaded(url, count, params=params)
```